platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java

```diff
@@ -463,6 +474,23 @@ private void wire() {
         selfEventSignerWiring
                 .getOutputWire()
                 .solderTo(internalEventValidatorWiring.getInputWire(InternalEventValidator::validateEvent), INJECT);
+        selfEventSignerWiring
```

Comment:
There is a (very unlikely) race condition present here

1. the `selfEventSigner` forwards a new event to the intake pipeline
2. is gossiped
3. comes to consensus
4. arrives in a consensus round at the `staleEventDetector`
5. goes ancient
6. And THEN gets passed into the`staleEventDetector`

There would be a false positive, where the event is declared stale, despite having come to consensus

Reply:
New architecture here: 

![Screenshot 2024-04-29 at 10 29 55 AM](https://github.com/hashgraph/hedera-services/assets/56973212/2ea28bf2-2849-4ae6-92ea-f62af9de7869)